libs/contrib/Text/Token.idr

```diff
@@ -0,0 +1,45 @@
+module Text.Token
+
+import Text.Lexer
+import Text.Parser
+
+%access public export
+%default total
+
+||| Convert a grammar with the token type `t` into one with a token type
+||| `TokenData t`.
+export
+wrapTokenData : Grammar t c a -> Grammar (TokenData t) c a
```

Comment:
This is included mostly as an example of how to use `mapToken`, but could be useful. I'm not a fan of this name either, though.

Reply:
I'm removing this.